lib/Myriad/Bootstrap.pm

```diff
@@ -78,6 +80,55 @@ sub allow_modules {
     @ALLOWED_MODULES{@_} = (1) x @_;
 }
 
+sub open_pipe {
+    # Establish comms channel for child process
+    socketpair my $child_pipe, my $parent_pipe, $constant{AF_UNIX}, $constant{SOCK_STREAM}, $constant{PF_UNSPEC}
+        or die $!;
+
+    { # Unbuffered writes
+        my $old = select($child_pipe);
+        $| = 1; select($parent_pipe);
+        $| = 1; select($old);
+    }
+
+    make_pipe_nonblocking($parent_pipe);
+    make_pipe_nonblocking($child_pipe);
+
+    return ($parent_pipe, $child_pipe);
+}
+
+sub make_pipe_nonblocking {
+    my $pipe = shift;
+    my $flags = fcntl($pipe, $constant{F_GETFL}, 0)
+        or die "Can't get flags for the socket: $!\n";
+
+    $flags = fcntl($pipe, $constant{F_SETFL}, $flags | $constant{O_NONBLOCK})
+        or die "Can't set flags for the socket: $!\n";
+}
+
+sub check_messages_in_pipe {
+    my ($pipe, $on_read) = @_;
+    # See https://perldoc.perl.org/functions/select#select-RBITS,WBITS,EBITS,TIMEOUT
+    # for a better understanding
+    my $input = '';
+    my $rin = my $win = '';
+    vec($rin, fileno($pipe), 1) = 1;
+    my $ein = $rin | $win;
+
+    die $! unless defined(my $nfound = select my $rout = $rin, my $wout = $win, my $eout = $ein, 0.5);
```

Comment:
This looks buggy. According to the documentation, `select` returns `-1` on error and sets `$!`. Maybe the documentation is wrong. But, alas:

```
> time perl -le 'my $bits=""; vec($bits, 15, 1)=1; my $ret=select $bits, undef, undef, 3; print "ret=$ret, !=$!"'
ret=-1, !=Bad file descriptor

real    0m0.003s
user    0m0.000s
sys     0m0.000s
```

It clearly didn't wait and it returned `-1` not `undef`.